packages/kotti-ui/source/kotti-filter/utils.ts

```diff
@@ -0,0 +1,152 @@
+import { useTranslationNamespace } from '../kotti-translation/hooks'
+import { Kotti } from '../types'
+
+export const getFilterEmptyValue = (filterKey: Kotti.Filter.Column['type']) => {
+	switch (filterKey) {
+		case Kotti.Filter.FilterType.BOOLEAN:
+		case Kotti.Filter.FilterType.NUMBER:
+		case Kotti.Filter.FilterType.SINGLE_ENUM:
+		case Kotti.Filter.FilterType.STRING:
+			return null
+		case Kotti.Filter.FilterType.DATE_RANGE:
+			return [null, null]
+		case Kotti.Filter.FilterType.MULTI_ENUM:
+			return []
+		default:
+			throw new Error('Invalid Filter Type')
+	}
+}
+
+export const getFilterInitialState = (
+	columnKey: Kotti.Filter.Column['key'],
+	columns: Kotti.Filter.Column[],
+): Kotti.Filter.InternalFilter => {
+	const column = columns.find((columnItem) => columnItem.key === columnKey)
+	switch (column?.type) {
+		case Kotti.Filter.FilterType.BOOLEAN:
+			return {
+				key: columnKey,
+				operation: Kotti.Filter.Operation.Boolean.EQUAL,
+				value: null,
+			}
+		case Kotti.Filter.FilterType.DATE_RANGE:
+			return {
+				key: columnKey,
+				operation: Kotti.Filter.Operation.DateRange.IN_RANGE,
+				value: [null, null],
+			}
+		case Kotti.Filter.FilterType.MULTI_ENUM:
+			return {
+				key: columnKey,
+				operation: Kotti.Filter.Operation.MultiEnum.ONE_OF,
```

Comment:
I think that special care needs to be taken to ensure that this `operation` is actually supported on `column.operations`